File: go-sdk/kestra_api_client/outputs_api.go
package kestra_api_client

import (
	"context"
)

type OutputsAPI struct {
	baseAPI
}

// ExecutionOutputs returns the flow-level outputs of an execution. Kestra 2.0
// dropped the `outputs` property from the execution payload itself, so this is
// the only way to read them.
func (a *OutputsAPI) ExecutionOutputs(ctx context.Context, executionId, tenant string) (map[string]interface{}, error) {
	return doJSON[map[string]interface{}](&a.baseAPI, ctx, "GET", tenantPath(tenant, "outputs", "executions", executionId), nil, nil)
}

// TaskOutputsInformation lists the task runs of an execution that have outputs.
func (a *OutputsAPI) TaskOutputsInformation(ctx context.Context, executionId, tenant string) ([]OutputControllerTaskOutputInformation, error) {
	return doJSON[[]OutputControllerTaskOutputInformation](&a.baseAPI, ctx, "GET", tenantPath(tenant, "outputs", "tasks", executionId), nil, nil)
}

// TaskRunOutputs returns the outputs of a single task run.
func (a *OutputsAPI) TaskRunOutputs(ctx context.Context, executionId, taskRunId, tenant string) (map[string]interface{}, error) {
	return doJSON[map[string]interface{}](&a.baseAPI, ctx, "GET", tenantPath(tenant, "outputs", "tasks", executionId, taskRunId), nil, nil)
}

File: python/python-sdk/kestrapy/api/outputs_api.py
from typing import Any, Dict, List

from kestrapy.base_api import BaseApi
from kestrapy.models.output_controller_task_output_information import (
    OutputControllerTaskOutputInformation,
)


class OutputsApi(BaseApi):
    """Execution outputs (`/api/v1/{tenant}/outputs`).

    Kestra 2.0 dropped the ``outputs`` property from the execution payload, so
    these routes are the only way to read flow- and task-level outputs.
    """

    def execution_outputs(self, execution_id: str, tenant: str) -> Dict[str, Any]:
        """The flow-level outputs of an execution.
        GET /api/v1/{tenant}/outputs/executions/{executionId}."""
        path = self._tenant_path(tenant, "outputs", "executions", execution_id)
        return self._raw_json_request("GET", path)

    def task_outputs_information(
        self, execution_id: str, tenant: str
    ) -> List[OutputControllerTaskOutputInformation]:
        """The task runs of an execution that have outputs.
        GET /api/v1/{tenant}/outputs/tasks/{executionId}."""
        path = self._tenant_path(tenant, "outputs", "tasks", execution_id)
        return self._json_list_request(
            "GET", path, OutputControllerTaskOutputInformation
        )

    def task_run_outputs(
        self, execution_id: str, task_run_id: str, tenant: str
    ) -> Dict[str, Any]:
        """The outputs of a single task run.
        GET /api/v1/{tenant}/outputs/tasks/{executionId}/{taskRunId}."""
        path = self._tenant_path(tenant, "outputs", "tasks", execution_id, task_run_id)
        return self._raw_json_request("GET", path)
